### ISSUE TITLE
Problem with ESP32 S3, When i updated arduino-esp32 and boards

### ISSUE THREAD

**Issue by motniemtin** [Spirik/GEM#97]
```
/libraries/GEM/src/GEMItem.cpp: In constructor 'GEMItem::GEMItem(const char*, byte&, GEMSelect&, void (*)(GEMCallbackData))':
/libraries/GEM/src/GEMItem.cpp:96:38: error: either all initializer clauses should be designated or none of them should be
   96 |   , callbackData{ .pMenuItem = this, { 0 } }
```

My demo Project https://wokwi.com/projects/402474219113695233

![image](https://github.com/Spirik/GEM/assets/23047848/925abbc1-d06a-4c9e-8cb5-8a5c3f1ca3ce)

**Comment by Spirik** [comment]
I was able to reproduce the issue after updating ESP32 boards via Arduino IDE. Will work on the fix next.

**Comment by motniemtin** [comment]
Hope you update soon

**Comment by Spirik** [comment]
Now fixed in release [1.5.3](https://github.com/Spirik/GEM/releases/tag/1.5.3).

The issue was due to more strict regulations of [designated initializers](https://mariusbancila.ro/blog/2020/02/27/c20-designated-initializers/) in recent compilers starting with C++20. And recent update of ESP32 core version 3.0 (that comes with boards update via Arduino IDE) uses C++23.

Wokwi demo compiles successfully now (you may need to remove and re-add GEM library for new version to be fetched):
<img width="1233" alt="image" src="https://github.com/Spirik/GEM/assets/1061406/3f153f45-8fb3-4283-974f-a76bbc64cc9e">

**Comment by motniemtin** [comment]
Thank you so much !!!
